### ISSUE TITLE
Test "protocol::generic_proto::tests::reconnect_after_disconnect" is flaky  

### ISSUE THREAD

**Issue by kianenigma** [paritytech/substrate#5574]
https://gitlab.parity.io/parity/substrate/-/jobs/454535#L2463

```
test protocol::generic_proto::tests::reconnect_after_disconnect ... FAILED
 failures:
 ---- protocol::generic_proto::tests::reconnect_after_disconnect stdout ----
 thread 'protocol::generic_proto::tests::reconnect_after_disconnect' panicked at 'explicit panic', client/network/src/protocol/generic_proto/tests.rs:397:55
 stack backtrace:
    0: backtrace::backtrace::libunwind::trace
              at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
    1: backtrace::backtrace::trace_unsynchronized
              at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
    2: std::sys_common::backtrace::_print_fmt
              at src/libstd/sys_common/backtrace.rs:77
    3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
              at src/libstd/sys_common/backtrace.rs:59
    4: core::fmt::write
              at src/libcore/fmt/mod.rs:1052
    5: std::io::Write::write_fmt
              at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/io/mod.rs:1426
    6: std::io::impls::<impl std::io::Write for alloc::boxed::Box<W>>::write_fmt
              at src/libstd/io/impls.rs:156
    7: std::sys_common::backtrace::_print
              at src/libstd/sys_common/backtrace.rs:62
    8: std::sys_common::backtrace::print
              at src/libstd/sys_common/backtrace.rs:49
    9: std::panicking::default_hook::{{closure}}
              at src/libstd/panicking.rs:204
   10: std::panicking::default_hook
              at src/libstd/panicking.rs:221
   11: std::panicking::rust_panic_with_hook
              at src/libstd/panicking.rs:472
   12: std::panicking::begin_panic
   13: <std::future::GenFuture<T> as core::future::future::Future>::poll
   14: std::thread::local::LocalKey<T>::with
   15: futures_executor::local_pool::block_on
   16: core::ops::function::FnOnce::call_once
   17: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
              at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/boxed.rs:1015
   18: __rust_maybe_catch_panic
              at src/libpanic_unwind/lib.rs:86
   19: std::panicking::try
              at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panicking.rs:281
   20: std::panic::catch_unwind
              at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/panic.rs:394
   21: test::run_test_in_process
              at src/libtest/lib.rs:539
   22: test::run_test::run_test_inner::{{closure}}
              at src/libtest/lib.rs:452
 note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
 failures:
     protocol::generic_proto::tests::reconnect_after_disconnect
```

**Comment by bkchr** [comment]
CC @tomaka

**Comment by tomaka** [comment]
Might be fixed after #5595. If not, might be fixed by https://github.com/libp2p/rust-libp2p/pull/1546

**Comment by kianenigma** [comment]
Closing for now since both PRs are merged. Let's see if it happens again.
